src/Validators/BaseValidator.php

```diff
@@ -0,0 +1,77 @@
+<?php
```

Comment:
The new validator component defines the workflow for how validation occurs and is processed.
"concrete" subclasses will override `configureValidator` and provide their validation rules.